src/spec/dsl.cr

```diff
@@ -225,6 +226,62 @@ module Spec
     end
   end
 
+  # :nodoc:
+  def self.execute_examples
+    log_setup
+    maybe_randomize
+    run_filters
+    root_context.run
+  end
+
+  # :nodoc:
+  def self.execute_list_tags
+    run_filters
+    tag_counts = collect_tags(root_context)
+    print_list_tags(tag_counts)
+    exit
```

Comment:
Is this `exit` used to skip the `finish_run` in the above `ensure`? If so I'd suggest not calling `finish_run` instead when `Spec.list_tags?` is true